src/Microsoft.AspNetCore.Razor.Design/build/netstandard2.0/Microsoft.AspNetCore.Razor.Design.targets

```diff
@@ -2,14 +2,16 @@
   <!-- 
     Targets supporting Razor MSBuild integration
   -->
+  <Import Project="Microsoft.AspNetCore.Razor.Design.Compilation.targets" />
+
   <Target Name="RazorGenerate" DependsOnTargets="$(RazorGenerateDependsOn)">
   </Target>
 
   <Target Name="RazorCompile" DependsOnTargets="$(RazorCompileDependsOn)">
   </Target>
 
   <PropertyGroup Condition="'$(RazorCompileOnBuild)'=='true'">
-    <BuildDependsOn>$(BuildDependsOn);RazorCompile</BuildDependsOn>
+    <PrepareForRunDependsOn>RazorCompile;$(PrepareForRunDependsOn)</PrepareForRunDependsOn>
```

Comment:
PrepareForRun is where they copy outputs to the output directory. Seems like a better target in case the user wants to do things with our outputs in `AfterBuild`

Reply:
love it

Reply:
Shouldn't we also attach to BuildDependsOn though? or does `Build` call `PrepareForRun`?

Reply:
CoreBuildDependsOn includes PrepareForRun. This is what the build graph looks like:

![image](https://user-images.githubusercontent.com/174281/34586618-8b2f98be-f159-11e7-81e2-03e28f316078.png)

Reply:
Looks good then.